core/src/main/java/org/apache/iceberg/BaseFileScanTask.java

```diff
@@ -217,6 +218,18 @@ public Iterable<FileScanTask> split(long splitSize) {
       throw new UnsupportedOperationException("Cannot split a task which is already split");
     }
 
+    @Override
+    public boolean isAdjacent(FileScanTask other) {
+      return file().equals(other.file()) && offset + len == other.start();
```

Comment:
I copied this logic from an existing place but we have to remember that files will be equal only if references are equal, which I think is true here. Our data and delete file implementations don't override equals.

Reply:
I think this is correct.

Reply:
In the other code we assume there is an ordering of the files. 

1 is adjacent to 2
2 is not adjacent to 1

We may want to change the api definition just to note that it is not commutative

Reply:
What about updating the code to cover both in the future?